apps/sms/views/shared/js/navigation.js

```diff
@@ -1,361 +1,921 @@
+ * The debug function translates to console.log in debug mode.
+ *
+ * @param {String} arg1 The first value to be displayed to the console.
+ * @param {...*} args The other values to be displayed to the console, or the
+ * parameters to the first string.
+ */
+var debug = 1 ?
+  (arg1, ...args) => console.log('[Navigation] ' + arg1, ...args) :
+  () => {};
+
+/**
+  * This contains an array of view characteristics, necessary to handle
+  * navigation requests.
+  * Each item has the following properties:
+  * - name: this is the name of the view. This will be used also in the URL
+  *   hash if the same HTML file holds several views.
+  * - url: this is the URL used to match the current location and find the
+  *   current view.
+  * - view: this is the actual JavaScript object that manages this view.
+  * - previous (optional): this is the name of the previous view; this is
+  *   necessary to decide if we do a forward/backward animation.
+  * - partOf (optional): this is used when a view is part of another view's
+  *   HTML document. In that case one view is "master" and the other are only
+  *   part of the master view.
+  * - alsoContains (optional): this is the opposite and holds the view's
+  *   subviews. (currently unused)
+  *
+  * NGA_VIEWS holds the configuration for NGA split views.
+  */
+const NGA_VIEWS = Object.freeze([
+  Object.freeze({
+    name: 'thread',
+    url: 'views/conversation/index.html',
+    behavior: 'ConversationView',
+    previous: 'thread-list',
+    alsoContains: ['report-view', 'composer', 'group-view']
+  }),
+  Object.freeze({
+    name: 'thread-list',
+    url: 'views/inbox/index.html',
+    behavior: 'InboxView'
+  }),
+  Object.freeze({
+    name: 'report-view',
+    behavior: 'ReportView',
+    previous: 'thread',
+    partOf: 'thread'
+  }),
+  Object.freeze({
+    name: 'composer',
+    behavior: 'ConversationView',
+    previous: 'thread-list',
+    partOf: 'thread'
+  }),
+  Object.freeze({
+    name: 'group-view',
+    behavior: 'GroupView',
+    previous: 'thread',
+    partOf: 'thread'
+  })
+]);
+
+/**
+  * OA_VIEWS holds the configuration for the old non-split architecture.
+  */
+const OA_VIEWS = Object.freeze([
+  Object.freeze({
+    name: 'thread-list',
+    url: 'index.html',
+    behavior: 'InboxView',
+    alsoContains: ['thread', 'composer', 'group-view', 'report-view']
+  }),
+  Object.freeze({
+    name: 'thread',
+    behavior: 'ConversationView',
+    previous: 'thread-list',
+    partOf: 'thread-list'
+  }),
+  Object.freeze({
+    name: 'composer',
+    behavior: 'ConversationView',
+    previous: 'thread-list',
+    partOf: 'thread-list'
+  }),
+  Object.freeze({
+    name: 'group-view',
+    behavior: 'GroupView',
+    previous: 'thread',
+    partOf: 'thread-list'
+  }),
+  Object.freeze({
+    name: 'report-view',
+    behavior: 'ReportView',
+    previous: 'thread',
+    partOf: 'thread-list'
+  })
+]);
+
+const VIEWS = isUsingOldArchitecture() ? OA_VIEWS : NGA_VIEWS;
```

Comment:
as a result we could pass made up views in tests too, so that we're not dependent on whatever views are defined here. This could be a follow-up